src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java

```diff
@@ -202,6 +207,45 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    void validatePath(FailureCollector collector) {
+      if (!containsMacro("path") && (!path.startsWith("s3a://") && !path.startsWith("s3n://"))) {
+        collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
+      }
+      if (!containsMacro("path") && path.startsWith("s3n://") && !Strings.isNullOrEmpty(connection.getSessionToken())) {
+        collector.addFailure("Temporary credentials are only supported for s3a:// paths.", null)
+          .withConfigProperty(NAME_PATH);
+      }
+    }
+
+    public void validateCredentials(FailureCollector collector, String path) {
+      AmazonS3 s3 = getS3Client(Regions.DEFAULT_REGION.getName());
+      try {
+        String bucketRegion = s3.getBucketLocation(S3Path.from(path).getBucket());
+        s3 = getS3Client(bucketRegion);
+        s3.getBucketAcl(S3Path.from(path).getBucket());
+      } catch (AmazonS3Exception exception) {
+        String errorMessage = exception.getErrorMessage();
+        collector.addFailure(String.format("Invalid values: %s.", errorMessage),
+                             "Please provide valid values.").withStacktrace(exception.getStackTrace());
+      }
+    }
+
+    private AmazonS3 getS3Client(String region) {
+      AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
+      if (region != null) {
+        builder.setRegion(region);
+      }
+      AWSCredentials creds;
+      if (connection.getSessionToken() == null) {
```

Comment:
Does this work in case the user is using IAM connection mode. If not put an early check to validate this flow only if required.

Reply:
Added validation check for the flow. https://github.com/data-integrations/amazon-s3-plugins/commit/a6e7bfaf0e9ff09084d865be1b6b75d2217d9267